### PR TITLE
docs: rest api guide

### LINKED ISSUE #158: Write a guide about REST api routes
A recent new feature of exposing REST on top of GraphQL was merged https://github.com/tailcallhq/tailcall/pull/1324

Explore this feature, and write a guide for everyone else to follow. Feel free to discuss this feature on the discord channel.

**IMPORTANT**
Make your writing engaging by adding LEGIBLE screenshots.

### PR DESCRIPTION
**Pull request by neo773**
This PR adds guide for `@rest` directive.

/claim #158
/closes #158

### CODE REVIEW

**Comment by tusharmath** [comment]
can you add `@rest` to the operators list.

**Comment by neo773** [comment]
@tusharmath 
Pushed

File: docs/operators/rest.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+---
+title: Rest API on top of GraphQL
```
Comment (tusharmath) [inline docs/operators/rest.md]:
```suggestion
title: REST on GraphQL
```

File: docs/operators/rest.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+This guide will show you how to use the `@rest` directive to create a RESTful service over a GraphQL layer for retrieving product information.
+
+## The @rest Directive Basics
+
+- **method**: Specifies the HTTP method (GET, POST, etc.).
+
+- **path**: Defines the endpoint URL, supporting dynamic values from query arguments.
+
+## Example: Retrieving Product Information
```
Comment (tusharmath) [inline docs/operators/rest.md]: Let's use a posts example from Jsonplaceholder. The original API when `/posts` is hit returns posts with only userId, here we should join it with the users API. That way when we hit `/posts` from tailcall, we get posts with some basic user information than just userId.

File: docs/operators/rest.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    id
+    title
+    price
+  }
+}
+```
+
+Let's try querying our newly created REST endpoint in our HTTP Client.
+We'll make a `GET` request to `http://127.0.0.1:8000/api/product/1` and see if we get the expected response.
+
```
Comment (tusharmath) [inline docs/operators/rest.md]: The summary should emphasize on the difference in the two APIs and the fact that
we uplifted the existing posts API and made it a lot more richer without writing any code.

File: docs/operators/rest.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+#### Response
+
+![REST Demo](/images/docs/rest.png)
```
Comment (tusharmath) [inline docs/operators/rest.md:76]: The image is great but should explain what's happening here. To summarize, a REST route is created dynamically from the provided GraphQL query/mutation. Whenever a request to `/api` is made, a matching route is looked up and it's GraphQL query is pulled up. This query is given to the graphql engine, the response is "RESTIFIED" and sent back. 

In better words :)

Comment (neo773) [inline docs/operators/rest.md]: @tusharmath 
Are you talking about the diagram image or the http client image?

File: docs/operators/rest.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+---
+title: REST on GraphQL
+description: Exposing REST endpoints on top of GraphQL via @rest directive.
+---
+
+Tailcall's `@rest` directive exposes standard REST endpoints, enabling the fetching of data from REST APIs directly through GraphQL queries.
+
+This guide will show you how to use the `@rest` directive to create a RESTful service over a GraphQL layer for retrieving product information.
```
Comment (tusharmath) [inline docs/operators/rest.md]: An intro about it's need:

Every one needs orchestration for APIs but everyone can't move to GraphQL and there are valid reasons for not moving to GraphQL. This feature allows people to building the Tailcall DSL, leveraging the GraphQL capabilities at compile time and generating REST endpoint for actual use instead. REST is a well-known standard so works with traditional API Infrastructure - CDNs Gateways etc.

File: docs/operators/rest.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+## The @rest Directive Basics
+
+- **method**: Specifies the HTTP method (GET, POST, etc.).
+
+- **path**: Defines the endpoint URL, supporting dynamic values from query arguments.
+
+## Example: Retrieving Product Information
+
+![Diagram](/images/docs/rest-diagram.svg)
```
Comment (tusharmath) [inline docs/operators/rest.md]: The image is great but should explain what's happening here. To summarize, a REST route is created dynamically from the provided GraphQL query/mutation. Whenever a request to /api is made, a matching route is looked up and it's GraphQL query is pulled up. This query is given to the graphql engine, the response is "RESTIFIED" and sent back.

In better words :)

File: docs/operators/rest.md (hunk 4 above)
Comment (tusharmath) [inline docs/operators/rest.md]: May be a closer screenshot?

File: docs/operators/rest.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
```
Comment (tusharmath) [inline docs/operators/rest.md]: Image still talks about products however the example talk about /posts and /users.

File: docs/operators/rest.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
+---
+title: "@rest"
```
Comment (tusharmath) [inline docs/operators/rest.md:2]: The guide and operator are two separate documents. I think we should keep it that way. What you have written is more of a guide than the operator documentation.

File: docs/operators/rest.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+Every one needs orchestration for APIs but everyone can't move to GraphQL and there are valid reasons for not moving to GraphQL. This feature allows people to building the Tailcall DSL, leveraging the GraphQL capabilities at compile time and generating REST endpoint for actual use instead. REST is a well-known standard so works with traditional API Infrastructure - CDNs Gateways etc.
+
+This guide will show you how to use the `@rest` directive to create a RESTful service over a GraphQL layer
+
+## The @rest Directive Basics
+
+- **method**: Specifies the HTTP method (GET, POST, etc.).
+
+- **path**: Defines the endpoint URL, supporting dynamic values from query arguments.
```
Comment (tusharmath) [inline docs/operators/rest.md]: We also have a `query` option to specify the query parameter. In the operator docs, we should explain how the DSL for `@rest` works. How the variables are inferred etc.

**Review by tusharmath — APPROVED** [review]

**Pull request merged**